src/pyhub/llm/agents/tools/code_interpreter/backends/docker.py

```
@@ -0,0 +1,428 @@
+    def cleanup(self):
+        """Cleanup any running containers."""
+        # Stop all code interpreter containers
+        for container in self.client.containers.list():
+            if container.name.startswith("code_interpreter_"):
+                try:
+                    container.stop()
+                    container.remove()
+                except:
+                    pass
+                    
+    def upload_file(self, session_id: str, local_path: Path, remote_path: Optional[str] = None) -> str:
```

Ruff (0.11.9)

Undefined name `Path`

(F821)